EMR: fix EBS configuration

cbamelis opened the pull request:
When I tried to create an EMR cluster with EBS volumes, my CloudFormation template yesterday failed.  After creating an AWS support ticket, it appeared the AWS documentation was wrong (EbsConfiguration cannot be specified on Cluster level, but needs to be specified in InstanceGroupConfigProperty).  The documentation has meanwhile been fixed and can be found on http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-emr-cluster-jobflowinstancesconfig-instancegroupconfig.html

The code below reflects this update.

rcuza commented on examples/EMR_Cluster.py:
```diff
@@ -90,7 +92,7 @@
         Name='Dummy bootstrap action',
         ScriptBootstrapAction=emr.ScriptBootstrapActionConfig(
-            Path='/bin/sh',
+            Path='file:/bin/sh',
```
How do you know the `file:` change is required? It makes sense but I don't see it in the AWS documentation on http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-emr-cluster-bootstrapactionconfig-scriptbootstrapactionconfig.html. When you stood up this template, did you verify that this worked?

rcuza commented:
I like the changes to the example script, especially how you imported the `troposphere.constants`. 

The changes in this PR move the EbcConfiguration into JobFlowInstancesConfig as per the AWS documentation. The PR updates the troposphere code, the example code, and the README. 

I was not able to build an EMR cluster using the example template in my account in US-WEST-2. 

```
09:50:40 UTC-0400 	ROLLBACK_IN_PROGRESS 	AWS::CloudFormation::Stack 	foo-emr-test 	The following resource(s) failed to create: [EMRSampleCluster]. . Rollback requested by user.
09:50:39 UTC-0400 	CREATE_FAILED 	AWS::EMR::Cluster 	EMRSampleCluster 	Encountered unsupported property EbsConfiguration
```
The example scripts should produce templates that work.

cbamelis commented:
I pushed a new commit with slightly cleaned up EMR_Cluster example (none of the parameters were actually used in the script).  I changed it to use the KeyPair and SubNet parameter and removed the unused security group parameters.  The stack is started successfully on my account in eu-west-1.  The hadoop step which was already present in the example fails (because it refers to an unexisting jar on S3, but I guess that's fine since the cloudformation stack is created successfully).

The CI build however failed on a curl download of python 3.5 (sounds unrelated to my changes).  Do I need to push another (artificial) change in order to trigger a new CI build? Or is there another way to retry the CI build?

rcuza commented:
The example script stands up a successfully. It creates the following resources:
```
AWS::IAM::InstanceProfile		
WS::IAM::Role		
AWS::EMR::Cluster		
AWS::IAM::Role		
AWS::EMR::Step
```

LGTM 👍

rcuza commented:
@markpeek - What is the best way to trigger the tests to run again? Should @cbamelis check in a minor commit or is there some other way to run them? Thank you.

markpeek commented:
Triggered build, successful run, and now merged. Thanks for the change.

@rcuza thanks for the code review and input.
